orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -208,6 +208,12 @@ def backward(ctx, *grad_outputs):
                     # Push user output grads to ONNX backend.
                     contiguous_grad_outputs = []
                     for idx, grad_output in enumerate(grad_outputs):
+                        if idx in self._onnx_graphs_info.output_grad_indices_non_differentiable:
```

Comment:
Is it possible to push this validation to the backend?

`self._onnx_graphs_info.output_grad_indices_non_differentiable` is populated in the backend and `grad_outputs` is passed in too

Reply:
This is actually a defensive programming practice...  
 
if grad_output is provided on non-differential-able outputs, the backend would crash on multiple places, e.g. the Yield's kernel assert on #gradient  == # Yield's output 
 
But I feel the best practice there to fail early, and provide the exact error message to user...